### PR TITLE
Merge Features to Dev branch-Code Review

### PR DESCRIPTION
**Pull request by MephistoDevelop**
Request to Code review with the next features:
-Create User Interface
-Start a new game
-Play a turn for each player
-Create Board with dynamical elements

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -15,15 +15,41 @@
...
+      <input type="button" onclick="giveName();" id="btn-play" value="play" />
+    </form>
+    <h2 id="messages">Message</h2>
+    <div class="container">
+      <div class="grid-container hide" id="board"></div>
+      <br />
+    </div>
+    <div id="buttons" class="hide">
+      <button id="button" onclick="resetGame();">Play Again</button>
+      <button id="button" onclick="refresh();">New Game</button>
```
Comment (mcrd25) [inline index.html]: You really ought to handle these onclicks in javascript script with an event handler.
Please do 🙏

File: index.html [hunk 2]
```diff
@@ -15,15 +15,41 @@
...
+        <li></li>
+      </ul>
+    </div>
+    <h1 class="head-text">Tic-Tac-Toe</h1>
+    <form id="form">
+      <div>
+        <input id="player-1" type="text" placeholder="Player 1" />
+        <input id="player-2" type="text" placeholder="Player 2" />
+      </div>
+      <input type="button" onclick="giveName();" id="btn-play" value="play" />
```
Comment (mcrd25) [inline index.html]: You really ought to handle this onclick in javascript script with an event handler.
Please do 🙏

File: index.html [hunk 3]
```diff
@@ -15,15 +15,41 @@
...
-
     <meta name="twitter:card" content="summary" />
     <meta name="twitter:site" content="@MephistoDevelop" />
     <meta name="twitter:site" content="@ansaryergesh " />
     <meta name="twitter:creator" content="@ansaryergesh " />
     <meta name="twitter:creator" content="@MephistoDevelop" />
+    <link type="text/css" href="assets/style.css" rel="stylesheet" />
   </head>
-  <body>
+  <body onload="resetGame();">
```
Comment (mcrd25) [inline index.html]: Try and handle this from the JavaScript file

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changes Required 👀 🔍 

Hi, @MephistoDevelop and @ansaryergesh 👋 ,

Your Tic Tac Toe interface looks good! And your README is excellent 👏!
However, there are still quite a few things to work on to make this project excellent and for it to be an even better addition to your portfolio.

### REquired Changes
- The goal of the project was to use as little global code as possible. As mentioned in the [Odin](https://www.theodinproject.com/courses/javascript/lessons/tic-tac-toe-javascript) description. You were actually supposed to encapsulate your logic into factories or modules as seen from this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/global_requirement.png). Please do this! 🙏 
- You need a game, gameBoard and player modules/factory. Hint 💡 player is best suited to be a factory.
- You ought to have a gameboard array (this could be in your gameBoard modules) and a function whose sole purpose is to render its contents as seen from this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/objects_requirement.png)
- Your winning method does not seem to compute the winner accurately as when I play I do not get correct feedback that I've won as seen from these screenshots. This is a requirement
Screen Shot 2020-01-13 at 2 12 20 AM
Screen Shot 2020-01-13 at 2 12 27 AM

### Suggested [OPTIONAL]
- Consider using ES6 syntax [OPTIONAL]

*Ultimately you ought to have at least 3 files. Your main app file and 3 files for your `game` module, `gameBoard` module and `player` factory. And you will import your modules/factory into your app.js (main file) and call methods there from those objects. Try to have as little code in your `app.js` as possible.*

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

File: dist/index.html [hunk 4]
```diff
@@ -15,15 +15,43 @@
...
+      <div class="grid-container hide" id="board"></div>
+      <br />
+    </div>
+    <div id="buttons" class="hide">
+      <button id="button1">Play Again</button>
+      <button id="button2">New Game</button>
+    </div>
     <script src="js/app.js"></script>
-    <div onclick="game()">Hello world</div>
+    <script type="module" src="js/board.js"></script>
```
Comment (Oghenebrume50) [inline dist/index.html:55]: Please where does the src attribute link to, I cannot see a js folder

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+  [2, 5, 8],
+  [3, 6, 9],
+  [1, 5, 9],
+  [3, 5, 7],
+];
+let winner;
+const options = ['X', 'O'];
+let turn = 0;
+let counter = 0;
+let messageText;
```
Comment (Oghenebrume50) [inline src/index.js:19]: You really do not need these globals screen

Creating reuasable objects is better, check out factory functions

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    } else {
+      player[0] = player1.value;
+    }
+
+    if (player2.value === '') {
+      player[1] = 'Player2';
+    } else {
+      player[1] = player2.value;
+    }
+    Board.drawBoard();
```
Comment (Oghenebrume50) [inline src/index.js:36]: [OPTIONAL] The player class should handle player logic, calling the board here to draw it, just shows how it doesn't do that very well

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
### STATUS: CHANGES REQUIRED 

Hi  Team, good job starting out on this and trying to fix the suggestions from previous review :+1: , please see more suggestions below  

I couldn't run this because the right javascript file was not linked to this project , please look into that 
**Please make changes and request for another review**

**RAPHAEL NORIODE** :sunglasses:

**Pull request closed without merging**